pkg/scaffold/olm-catalog/csv.go

```diff
@@ -0,0 +1,326 @@
+		currCSV = new(olmApi.ClusterServiceVersion)
+		s.initCSVFields(currCSV)
+	}
+
+	csvConfig, err := getCSVConfig(s.ConfigFilePath)
+	if err != nil {
+		return nil, err
+	}
+
+	if err = s.updateCSVVersions(currCSV); err != nil {
+		return nil, err
+	}
+	if err = s.updateCSVFromManifestFiles(currCSV, csvConfig); err != nil {
+		return nil, err
+	}
+
+	// A new csv won't have several required fields populated.
+	if err = checkRequiredCSVFields(currCSV); err != nil {
+		if exists {
+			log.Warnf("required csv fields not filled in file %s:%s\n", s.Path, err)
+		} else {
+			// Report required fields to user informationally.
+			log.Infof("fill in the following required fields in file %s:%s\n", s.Path, err)
+		}
+	}
+
+	return yaml.Marshal(currCSV)
+}
+
+func getCurrentCSVIfExists(csvPath string) (*olmApi.ClusterServiceVersion, bool, error) {
+	if _, err := os.Stat(csvPath); err != nil && os.IsNotExist(err) {
+		return nil, false, nil
+	}
+
+	csvBytes, err := ioutil.ReadFile(csvPath)
+	if err != nil {
+		log.Infof("getCurrentCSVIfExists ReadFile: (%v)", err)
+		return nil, false, err
+	}
+	if len(csvBytes) == 0 {
+		return nil, false, nil
+	}
+
+	csv := new(olmApi.ClusterServiceVersion)
+	if err := yaml.Unmarshal(csvBytes, csv); err != nil {
+		log.Infof("getCurrentCSVIfExists Unmarshal: (%v)", err)
+		return nil, false, err
+	}
+
+	return csv, true, nil
+}
+
+func getCSVName(name, version string) string {
+	return name + ".v" + version
+}
+
+// getDisplayName turns a project dir name in any of {snake, chain, camel}
+// cases, hierarchical dot structure, or space-delimited into a
+// space-delimited, title'd display name.
+// Ex. "another-_AppOperator_againTwiceThrice More"
+// ->  "Another App Operator Again Twice Thrice More"
+func getDisplayName(name string) string {
+	for _, sep := range ".-_ " {
+		splitName := strings.Split(name, string(sep))
+		for i := 0; i < len(splitName); i++ {
+			splitName[i] = strings.TrimSpace(splitName[i])
+		}
+		name = strings.Join(splitName, " ")
+	}
+	splitName := strings.Split(name, " ")
+	for i, word := range splitName {
+		temp := word
+		o := 0
+		for j, r := range word {
+			if unicode.IsUpper(r) {
+				if j > 0 && !unicode.IsUpper(rune(word[j-1])) {
+					temp = temp[0:j+o] + " " + temp[j+o:len(temp)]
+					o++
+				}
+			}
+		}
+		splitName[i] = temp
+	}
+	return strings.TrimSpace(strings.Title(strings.Join(splitName, " ")))
+}
+
+// initCSVFields initializes all csv fields that should be populated by a user
+// with sane defaults. initCSVFields should only be called for new csv's.
+func (s *Csv) initCSVFields(csv *olmApi.ClusterServiceVersion) {
+	// Metadata
+	csv.TypeMeta.APIVersion = olmApi.ClusterServiceVersionAPIVersion
+	csv.TypeMeta.Kind = olmApi.ClusterServiceVersionKind
+	csv.SetName(getCSVName(strings.ToLower(s.ProjectName), s.CsvVersion))
+	csv.SetNamespace("placeholder")
+
+	// Spec fields
+	csv.Spec.Version = *semver.New(s.CsvVersion)
+	csv.Spec.DisplayName = getDisplayName(s.ProjectName)
```

Comment:
Can `DisplayName` be an empty string? Should we add a validation for that?

Reply:
DisplayName will be validated by `checkRequiredCSVFields`